ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts/templates.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible_collections.arista.avd.plugins.module_utils.utils import get, default
+
+
+# List of keys read from vars and overloaded with "node_type_keys.<node_type_data_key>" and then returned as dict.
+TEMPLATE_FACTS = [
+    {
+        "set": "ip_addressing",
+        "get": "templates.ip_addressing",
+        "node_type_data_key": "ip_addressing",
+        "default": {}
+    }, {
+        "set": "interface_descriptions",
+        "get": "templates.interface_descriptions",
+        "node_type_data_key": "interface_descriptions",
+        "default": {}
+    }
+]
+
+
+def get_template_facts(vars, filters):
+    """
+    Get General switch.* facts based on the "node_type_keys" and "templates" data models
+
+    These facts are pointers to j2 templates used for ip_addressing and interface descriptions.
+
+    Parameters
+    ----------
+    vars : dict
+        Hostvars for one host
+    filters : dict
+        Jinja filters loaded in main module
+
+    Returns
+    -------
+    dict
+        switch.*: Facts based on the "node_type_keys" and "templates" data models
+    """
+
+    switch = vars['switch']
+    combine = filters['combine']
+
+    node_type_key = switch['node_type_key']
+
+    for template_fact in TEMPLATE_FACTS:
+        get_key = template_fact['get']
+        set_key = template_fact['set']
+        node_type_data_key = template_fact['node_type_data_key']
+        default_value = template_fact['default']
+
+        var_templates = get(vars, get_key, {})
+        node_type_templates = get(vars, f"node_type_keys.{node_type_key}.{node_type_data_key}", {})
```

Comment:
The contents of `e` should still lead us on the way. We could make our error message instead of relying on `KeyError`, and then we catch on that in the wrapper function. We could use our own `get` with `required=True` everywhere to make both ensure the correct error is raised, but also to make it clear to the reader that this is a datamodel variable we have as `required`.